disperser/common/blobstore/shared_storage.go

```diff
@@ -158,6 +160,17 @@ func (s *SharedBlobStore) MarkBlobConfirmed(ctx context.Context, existingMetadat
 }
 
 func (s *SharedBlobStore) MarkBlobDispersing(ctx context.Context, metadataKey disperser.BlobKey) error {
+	refreshedMetadata, err := s.GetBlobMetadata(ctx, metadataKey)
+	if err != nil {
+		s.logger.Error("error getting blob metadata while marking blobDispersing", "err", err)
+		return err
+	}
+	alreadyConfirmed, _ := refreshedMetadata.IsConfirmed()
```

Comment:
What about just checking if the status is PROCESSING? That's the only valid predecessor for DISPERSING.

Reply:
updated